addons/web/static/src/core/emoji_picker/emoji_picker.js

```diff
@@ -306,6 +312,9 @@ export class EmojiPicker extends Component {
     }
 
     getEmojis() {
+        if (this.recentEmojis.length > 0 && this.searchTerm.length > 1) {
+            this.emojis = this.emojis.filter((emoji) => !this.recentEmojis.includes(emoji));
+        }
```

Comment:
IMO `emojis` should not be updated like this. If you need you can create a local copy and than update it for your use. The main object should not be altered.